### ISSUE TITLE
Failed validation does not return an error code

### ISSUE THREAD

**Issue by mboersma** [Azure/oav#191]
Command-line tools that always return 0 for exit status aren't POSIX-compliant and can't easily be used in scripting.

```
$ oav --version
0.4.21
$ oav validate-example Microsoft.ContainerService/2017-01-31/containerService.json 
error:  Error: In operation "ContainerServices_List", exampleParameterValues cannot be null or undefined and must be of type "object" (A dictionary of key-value pairs of parameter-names and their values).
    at SpecValidator.validateRequest (/usr/local/lib/node_modules/oav/lib/validators/specValidator.js:566:13)
    at SpecValidator.validateXmsExamples (/usr/local/lib/node_modules/oav/lib/validators/specValidator.js:522:60)
    at SpecValidator.validateOperation (/usr/local/lib/node_modules/oav/lib/validators/specValidator.js:464:10)
    at SpecValidator.validateOperations (/usr/local/lib/node_modules/oav/lib/validators/specValidator.js:498:12)
    at /usr/local/lib/node_modules/oav/lib/validate.js:108:15
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:678:11)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
(node:2310) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: In operation "ContainerServices_List", exampleParameterValues cannot be null or undefined and must be of type "object" (A dictionary of key-value pairs of parameter-names and their values).
(node:2310) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
$ echo $?
0
```

**Comment by amarzavery** [comment]
@mboersma - This has been fixed in 0.4.22. Please feel free to reopen this issue if the problem persists.

**Comment by mboersma** [comment]
@amarzavery I tested with 0.4.22 and `oav` seems to be returning 1 when the command is mis-typed or the validation command fails. Thanks for fixing this so quickly!
